projects/js-packages/publicize-components/src/components/social-post-modal/settings-section.tsx

```diff
@@ -64,6 +66,18 @@ export function SettingsSection( { onReShared } ) {
 					) }
 				</p>
 				<SharePostForm analyticsData={ { location: 'preview-modal' } } />
+				{ needsUserConnection && (
+					<Notice type={ 'warning' }>
+						{ __(
+							'You must connect your WordPress.com account to be able to re-share posts.',
+							'jetpack-publicize-components'
+						) }
+						&nbsp;
+						<a href={ getMyJetpackUrl( '#/connection' ) }>
+							{ __( 'Connect now', 'jetpack-publicize-components' ) }
+						</a>
+					</Notice>
+				) }
```

Comment:
I am a bit confused here. It means that they can't re-share to shared connections if they haven't connected their account, but if they connect, they can, right?

If that is the case, why do we force them to connect their accounts? What problem are we trying to solve?

Reply:
> It means that they can't re-share to shared connections if they haven't connected their account, but if they connect, they can, right?

Yes.

> What problem are we trying to solve?

The problem is that resharing is not possible without user conneciton, but the UI lets you click the button and eveything gets messed up.

<img width="2938" height="2026" alt="CleanShot 2025-07-29 at 10 05 44 png" src="https://github.com/user-attachments/assets/3675f9d5-0266-4dff-b6d0-b3a636e73cc1" />

Reply:
That makes sense. Thanks.